### ISSUE TITLE
SC1001: false positive on \:

### ISSUE THREAD

**Issue by ajeetdsouza** [koalaman/shellcheck#2287]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC1001
- My shellcheck version (`shellcheck --version` or "online"): 0.7.2
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/sh
\:
```

#### Here's what shellcheck currently says:

```
Line 2:
\:
^-- SC1001: This \: will be a regular ':' in this context.
```

#### Here's what I wanted or expected to see:

This is not an error. In a POSIX shell eg. `dash`, it's possible to define `alias ':'=exit`, when `:` is actually supposed to mean `true`. Using `\:` allows me to unambiguously refer to the correct definition of `:`.

**Comment by koalaman** [comment]
ShellCheck recognized this idiom for e.g. `\ls` but not for symbols like `:` or `.`. These have now been added to the list. Thanks!

(For especially obscure alias names, the name can also be double quoted as in "%")
